api/src/main/java/com/velocitypowered/api/proxy/Player.java

```diff
@@ -27,6 +27,13 @@
      */
     UUID getUniqueId();
 
+    /**
+     * Sets the player's UUID.
+     * Can be set only in Login state
+     * @param uuid
+     */
+    void setUniqueId(UUID uuid);
```

Comment:
During the login process, the `Player` object is not available. It only becomes available _after_ the login process has completed (in other words, when `LoginEvent` is fired).